sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -56,7 +64,7 @@ class AzureAppConfigurationClient:
     """
 
     # pylint:disable=protected-access
-    def __init__(self, base_url: str, credential: TokenCredential, **kwargs) -> None:
+    def __init__(self, base_url: str, credential: TokenCredential, *, api_version: str = "1.0", **kwargs) -> None:
```

Comment:
Do we want to show this kwarg in intellisense?

Reply:
Why not? 1. more clear to show customers that `api_version` is configurable in api  2. checked other languages, e.g. in .Net, [`api_version`](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClientOptions.cs#L45) is showing up in their apiview

Reply:
Other Python SDKs do not explicitly show this kwarg.

Reply:
ACR does([link](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py#L68)), I remember it was requested to add. I can check whether to expose it here with v-team.

Reply:
Based on the api [guideline](https://azure.github.io/azure-sdk/general_design.html#service-api-versions): 
"The purposes of the client library is to communicate with an Azure service. Azure services support multiple API versions. To understand the capabilities of the service, the client library **must** be able to support multiple service API versions."
I think it is required to support multiple api versions. @xiangyan99 Do you know why other packages choose not to?

Reply:
Took offline.

Reply:
Reverted the surface updates here, and target the latest `api_version` to `2022-11-01-preview` as it include snapshot operations.